src/Orchard.Web/Modules/Orchard.OutputCache/Filters/OutputCacheFilter.cs

```diff
@@ -206,8 +199,17 @@ public class OutputCacheFilter : FilterProvider, IActionFilter, IResultFilter, I
                 var cacheDuration = _cacheRouteConfig != null && _cacheRouteConfig.Duration.HasValue ? _cacheRouteConfig.Duration.Value : CacheSettings.DefaultCacheDuration;
                 var cacheGraceTime = _cacheRouteConfig != null && _cacheRouteConfig.GraceTime.HasValue ? _cacheRouteConfig.GraceTime.Value : CacheSettings.DefaultCacheGraceTime;
 
-                // Include each content item ID as tags for the cache entry.
-                var contentItemIds = _displayedContentItemHandler.GetDisplayed().Select(x => x.ToString(CultureInfo.InvariantCulture)).ToArray();
+                // Get the tags for this cache item from.
+                var cacheItemTags = new List<string>();
+
+                foreach (var cacheTagProvider in _cacheTagProviders) {
+                    try {
+                        cacheItemTags.AddRange(cacheTagProvider.GetTags());
+                    }
+                    catch (Exception ex) {
+                        Logger.Warning(ex, "Cache tags from provider {0} will not be added to the cached item ({1}) because the provider threw an exception when asked to provide tags.", cacheTagProvider.GetType().FullName, _cacheKey);
```

Comment:
An individual provider failing should not prevent the request from succeeding. Log warn and continue.